app/services/idv/inherited_proofing/va/mocks/service.rb

```diff
@@ -35,17 +35,22 @@ class Service
             },
           }.freeze
 
+          ERROR_HASH = {
+            errors: 'InheritedProofing::Errors::MHVIdentityDataNotFoundError',
+          }.freeze
+
           def initialize(service_provider_data)
             @auth_code = service_provider_data[:auth_code]
           end
 
           def execute
-            if (@auth_code != VALID_AUTH_CODE)
-              raise TypeError,
-                    "auth_code is invalid: #{@auth_code}"
-            end
+            valid_auth_code ? PAYLOAD_HASH : ERROR_HASH
+          end
+
+          private
 
-            PAYLOAD_HASH
+          def valid_auth_code
+            @auth_code == VALID_AUTH_CODE
```

Comment:
I would invert this. Most of our other mock proofers treat everything as valid, except for some specific error codes, to get specific error responses, that we document. See the phone numbers here: https://developers.login.gov/testing/#phone-number-verification

Reply:
Makes my skin crawl, but change has been made.

Reply:
Thanks for making updates. I think we got some wires crossed?

This part of the new code:
```ruby
          def invalid_auth_code
            @auth_code != VALID_AUTH_CODE
          end
```

Is that the part you're getting at? What I was trying to suggest is that we have a single, fixed `INVALID_AUTH_CODE` and then everything else is valid

Reply:
Hi @zachmargolis I think @holytoastr was trying to achieve consistency between how the IP process works, regardless of whether or not we are making mock or live calls to the VA staging server. From the beginning, the VA has given us _1 valid auth_code_ to use; any other auth_codes passed along to the VA staging server are considered invalid. So, we have consistency between how IP behaves in, for example, the staging environment - regardless of whether or not we're using the mock or calling the live VA staging server. IP has also been, up to now, feverishly (borderline obsessively) demoed to stakeholders exclusively in the staging environment. Depending on what we are demoing and where we are at with our codebase, it may require us to turn the mock on/off depending - we need the consistency. Demoing on the staging server requires the sinatra app, which does not allow us to enter auth_codes manually (they are chosen from a select list). Why a select list and not user-enterable? There is no need at the moment for IP to consider any other auth_codes valid except the one the VA provided to us. Any other auth_codes sent to IP will be considered _invalid_ - consistent with the VA staging server. Consequently, inversion of the mock code in question is somewhat of an inversion of our needs at the moment. Remember, this is a _VA mock proofer_. Other service providers in the future may do things differently, and it may certainly make more sense to take an "open gate" approach to auth_codes, rather than the approach taken here. As it stands, this fits pretty well with what _VA_ IP needs, I think this may be more of a preference since you used the word _"suggest"_? I'm trying to see the smell, but I'm not seeing it. If this is all that's holding up this code from getting merged, I'm not sure this is worth holding it up. Let us know your thoughts.

Reply:
That seems like an ok reason to do the original approach for now, but let's definitely consider circling back to make things a consistent experience later!

Reply:
Just for bigger picture context, since the auth code is provided by the VA when the user is sent to us we don't have a place where we would input the code the way you would with a phone number. In the end, I am envisioning a response like below once we have more than 2 auth_codes to work with. However, since we're agile, we're keeping it simple for now with a tertiary operator and will iterate as new codes are provided to us by the devs working on the VA api.

```ruby
PAYLOAD_HASH = {...}
ERROR_HASH_A = {...}
ERROR_HASH_B = {...}
GENERIC_ERROR_HASH = {...}
PAYLOAD_HASHES = {
  Idv::InheritedProofing::Va::Mocks::Service::VALID_AUTH_CODE => PAYLOAD_HASH,
  'error_a_auth_code' => ERROR_HASH_A,
  'error_b_auth_code' => ERROR_HASH_B,
}

PAYLOAD_HASHES[auth_code] || GENERIC_ERROR_HASH
```

Hopefully you find that helpful and can approve this PR to help move it along